src/workers.rs

```diff
@@ -66,20 +67,20 @@ impl Worker for Sidekiq {
         "sidekiq"
     }
 
-    fn work_to_do(&self, redis_conn: &redis::Connection) -> bool {
-        let queues: Vec<String> = redis_conn.keys(format!("{}:queue:*", self.namespace)).expect("Could not read from redis");
+    fn work_to_do(&self, redis_conn: &redis::Connection) -> Result<bool, FromanError> {
+        let queues: Vec<String> = redis_conn.keys(format!("{}:queue:*", self.namespace))?;
         let counts: Vec<i32> = queues.iter().map(|q| {
-            redis_conn.llen(q).unwrap()
+            redis_conn.llen(q).unwrap_or(0)
         }).collect();
-        counts.iter().fold(0i32, |a, &b| a + b) > 0
+        Ok(counts.iter().fold(0i32, |a, &b| a + b) > 0)
```

Comment:
can [sum](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum) work? maybe not because of that reference?

Reply:
I'm not sure if this is better?

```rust
        let counts: Result<Vec<i32>, _> = queues.iter().map(|q| {
            redis_conn.llen(format!("{}:queue:{}", self.namespace, q))
        }).collect();
        Ok(counts?.iter().sum::<i32>() > 0)
```

Reply:
[here](https://rustbyexample.com/error/iter_result.html#fail-the-entire-operation-with-collect) is what it is doing

Reply:
or at least what I _think_ it is doing 😆